Source/WebCore/svg/animation/SVGSMILElement.cpp

```diff
@@ -882,15 +882,14 @@ void SVGSMILElement::resolveInterval(bool first, SMILTime& beginResult, SMILTime
     
 void SVGSMILElement::resolveFirstInterval()
 {
-    SMILTime begin;
-    SMILTime end;
-    resolveInterval(true, begin, end);
-    ASSERT(!begin.isIndefinite());
+    SMILInterval firstInterval;
+    resolveInterval(true, firstInterval.begin, firstInterval.end);
+    ASSERT(!firstInterval.begin.isIndefinite());
 
-    if (!begin.isUnresolved() && (begin != m_intervalBegin || end != m_intervalEnd)) {   
+    if (!firstInterval.begin.isUnresolved() && firstInterval != SMILInterval(m_intervalBegin, m_intervalEnd)) {
```

Comment:
The fix could just be to add `.value()` here. Adding struct is unnecessary and also slightly dubious semantically since it’s not all that clear to have it just happen to implement `!=` differently.

Just `begin.value() != m_intervalBegin.value() || end.value() != m_intervalEnd.value()`.